src/sys/socket/sockopt.rs

```diff
@@ -513,23 +514,23 @@ unsafe impl<T> Get<CString> for GetCString<T> {
         &mut self.len
     }
 
-    unsafe fn unwrap(mut self) -> CString {
-        CStr::from_ptr(self.ffi_ptr() as *const c_char).to_owned()
+    unsafe fn unwrap(self) -> OsString {
+        OsStr::from_bytes(self.val.as_ref()).to_owned()
     }
 }
 
-/// Setter for a `CString` value.
-struct SetCString<'a> {
-    val: &'a CString,
+/// Setter for a `OsString` value.
+struct SetOsString<'a> {
+    val: &'a OsString,
 }
 
-unsafe impl<'a> Set<'a, CString> for SetCString<'a> {
-    fn new(val: &'a CString) -> SetCString {
-        SetCString { val }
+unsafe impl<'a> Set<'a, OsString> for SetOsString<'a> {
```

Comment:
I think this entire `impl` block doesn't need to be `unsafe`.

Reply:
You're right, but this is defined by the trait
```rust
unsafe trait Set<'a, T>
```

Reply:
Hm, that's a mistake.  A Trait should only be marked as `unsafe` if the `impl` itself can cause unsafe behavior, not just because it has unsafe methods.  For example, `Send`, and `Sync` are unsafe methods.  It's fine if you want to leave that in for this PR, but we should remove it in another PR.